src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -525,6 +527,18 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks a record component definition. Record components are private
+     * final fields, so we do not consider memberModifiers field.
+     *
+     * @param recordComponentDef the record component definition to check.
+     */
+    public void visitRecordComponentDef(DetailAST recordComponentDef) {
+        if (recordComponentDef.getType() == TokenTypes.RECORD_COMPONENT_DEF) {
```

Comment:
Explain why we need this condition.

Reply:
We know that the RECORD_COMPONENT_DEF can catch illegal types from any place so we need a condition that can check to be it public, private, protected, final, We need a check which can check RECORD_COMPONENT-DEF type.

So if  whatever the type is of recordComponentDef, if it is equal to the token, we defined, then checkClassName(recordComponentDef);

Reply:
Let me rephrase; how is this condition different than the precondition required to reach this method?

Reply:
In previous conditions, we were specifically checking for private and final member modifiers but Romani sir said that It not only check private and final but anything. So we have to consider them like constructors.

This condition is different from the pre-defined condition because before we were only checking the fields of the record components but now with this condition, we are checking every field and not specific to any particular field.

Reply:
So we had an option to how to treat these member modifiers and with your options that we can check public or private/final or we can check them for all the specified memberModifiers property

Reply:
Please guide me If It is wrong.

Reply:
Please take some time, and read the documentation for `DetailAST#getType` and `TokenTypes`. Then, generate a coverage report https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations to see what line is failing if you have not done this already. After that, try to understand why the code that is not covered is failing.

Reply:
Sure, thanks for the link.
@nmancus1 but every line is covered when I ran code coverage in my local environment. It's 100%

Reply:
Then your local branch is not in sync with this PR branch.

Reply:
@nmancus1 I think this will work better?
 ```
 public void visitRecordComponentDef(DetailAST recordComponentDef) {
       checkClassName(recordComponentDef);
    }
```

Reply:
Resolved